### PR TITLE
[trial_build] Fix bugs

### PR DESCRIPTION
**Pull request by jonathanmetzman**
1. Don't try to build the script name as a project.
2. Add a flag to force builds of projects that previously failed.
3. Make sure we build projects from our PR branch.
4. Wait on all builds not just builds of the last type.
5. Don't use test bucket for corpus or coverage (will corpus work or fail because of creds?)

Add tests for these features.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py aiohttp --sanitizer coverage address --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py aiohttp --sanitizer coverage address --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py aiohttp --sanitizer coverage address --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py aiohttp --sanitizer coverage address --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
FYI the infra tests breakage should hopefully be fixed by https://github.com/google/oss-fuzz/pull/7766

**Comment by oliverchang** [comment]
Is this ready for review? I think there's just a lint/format issue in presubmit :)

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py aiohttp --sanitizer coverage address --fuzzing-engine libfuzzer

File: infra/build/functions/build_lib.py [hunk 1]
```diff
@@ -105,8 +105,11 @@ def get_upload_bucket(engine, architecture, testing):
   return bucket
 
 
-def _get_targets_list(project_name, testing):
+def _get_targets_list(project_name):
   """Returns target list."""
+  # We never want the target list from the testing bucket, the testing bucket is
+  # only for uploading.
+  testing = None
```
Comment (oliverchang) [inline infra/build/functions/build_lib.py:112]: Just set `testing=None` in the call arguments on line 115?

**Review by oliverchang — APPROVED** [review]

File: infra/base-images/tmp [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply infra/base-images/tmp]: Yeah it was so that gcb would test this PR.

**Pull request merged**